crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -302,11 +303,9 @@ pub(super) fn extract(s: &Series, pat: &str, group_index: usize) -> PolarsResult
 
 #[cfg(feature = "extract_groups")]
 /// Extract all capture groups from a regex pattern as a struct
-pub(super) fn extract_groups(s: &Series, pat: &str) -> PolarsResult<Series> {
-    let pat = pat.to_string();
```

Comment:
No need to allocate a string here?

Reply:
Yep, thanks. Need to become more aware regarding allocations.